### PR TITLE
CAT-1301_permissions

### PR DESCRIPTION
**Pull request by X4mp**
implemented project.getRequiredRessources
implemented generatePermissionListFromRessource
called in StorageHandler.saveProject

changed Filename and path

### CODE REVIEW

**Comment by robertpainsi** [comment]
http://thecodinglove.com/post/111849339882/when-i-find-a-hard-coded-string-during-code-review

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java [hunk 1]
```diff
@@ -393,6 +395,14 @@ public boolean saveProject(Project project) {
 			writer = new BufferedWriter(new FileWriter(tmpCodeFile), Constants.BUFFER_8K);
 			writer.write(projectXml);
 			writer.flush();
+
+			FileWriter permissionWriter = new FileWriter(buildProjectPath(project.getName()) + PROJECTPERMISSIONS_NAME);
+
+			for(String str: project.getRequiredResources()) {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: Rename please!

File: catroid/src/org/catrobat/catroid/content/Project.java [hunk 2]
```diff
@@ -136,6 +136,39 @@ public XmlHeader getXmlHeader() {
 		return this.xmlHeader;
 	}
 
+	public List<String> getRequiredResources() {
+		int resources = Brick.NO_RESOURCES;
+
+		for (Sprite sprite : spriteList) {
+			resources |= sprite.getRequiredResources();
+		}
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/Project.java]: https://github.com/Catrobat/Catroid/blob/master/catroid/src/org/catrobat/catroid/stage/PreStageActivity.java#L313-L322
Looks like the same code to me. Please refactor

Btw: Not sure why the PreStageActivity has a member called resources. Maybe it's needed in the tests... https://github.com/Catrobat/Catroid/blob/master/catroid/src/org/catrobat/catroid/stage/PreStageActivity.java#L75

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java (hunk 1 above)
Comment (X4mp) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: rename what?

File: catroid/src/org/catrobat/catroid/content/Project.java (hunk 2 above)
Comment (X4mp) [inline catroid/src/org/catrobat/catroid/content/Project.java]: looks like the same code to YOU, but isn't

**Comment by X4mp** [comment]
please test this

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java (hunk 1 above)
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: `str` to something that isn't an abbreviation, like `resource` would be my guess.

**Comment by X4mp** [comment]
please test this

**Comment by X4mp** [comment]
test this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by X4mp** [comment]
retest this please

**Comment by linsm** [comment]
please reopen your pull request 
 - use your forked repository
 - target of pull request should be 'develop' branch
 - link THIS pull request in the new one

**Pull request closed without merging**